pkg/provisioner/conditionwaiter.go

```diff
@@ -59,9 +55,8 @@ func (w *conditionWaiter) Wait(ctx context.Context) (runtime.Object, error) {
 	// if not wait for the event handler signal
 	select {
 	case <-ctx.Done():
-		errMsg := fmt.Sprintf("context timeout for %s (%s)", w.objType, w.objName)
-		klog.Error(errMsg)
-		return nil, status.Errorf(codes.DeadlineExceeded, errMsg)
+		klog.Errorf("context timeout for %s (%s)", w.objType, w.objName)
+		return nil, context.DeadlineExceeded
```

Comment:
We should really return `ctx.Err()` here. The context may have been canceled in which case the error would be `context.Canceled`.

Reply:
Done. Also added condition for `errors.Is(err, context.Canceled)` in `crdProvisioner.CreateVolume`.